### PR TITLE
Refactor inverted indexes instances

### PR DESCRIPTION
**Pull request by wmitsuda**
This is a 1 of N refactoring, I think there is room to improve the code even more, but this PR is already big, so for now just removing duplicated code and grouping some abstractions.

This PR does NOT change any code semantics.

- Remove lots of duplicated code.
- Makes the internal management of the inverted indexes which are not associated with domains more like how domains are arranged.
- Main driver here is to make the code easier to add more inverted indexes, the current code requires touching a lot of places and encourage copy/pasting and repeating code.

### CODE REVIEW

**Review by AskAlexSharov — APPROVED** [review]

File: erigon-lib/state/aggregator.go [hunk 1]
```diff
@@ -180,20 +177,16 @@ func NewAggregator(ctx context.Context, dirs datadir.Dirs, aggregationStep uint6
 	//if a.d[kv.GasUsedDomain], err = NewDomain(cfg, aggregationStep, "gasused", kv.TblGasUsedKeys, kv.TblGasUsedVals, kv.TblGasUsedHistoryKeys, kv.TblGasUsedHistoryVals, kv.TblGasUsedIdx, logger); err != nil {
 	//	return nil, err
 	//}
-	idxCfg := iiCfg{salt: salt, dirs: dirs, db: db}
-	if a.logAddrs, err = NewInvertedIndex(idxCfg, aggregationStep, "logaddrs", kv.TblLogAddressKeys, kv.TblLogAddressIdx, nil, logger); err != nil {
+	if err := a.registerII(kv.LogAddrIdxPos, salt, dirs, db, aggregationStep, "logaddrs", kv.TblLogAddressKeys, kv.TblLogAddressIdx, logger); err != nil {
```
Comment (AskAlexSharov) [inline erigon-lib/state/aggregator.go:180]: I think can use `kv.LogAddrIdxPos.String()` instead of `"logaddrs"`

Comment (wmitsuda) [reply erigon-lib/state/aggregator.go:180]: unfortunately not, it’s a number constant

Comment (AskAlexSharov) [reply erigon-lib/state/aggregator.go:180]: @wmitsuda see Domain type. It has String() method

Comment (AskAlexSharov) [reply erigon-lib/state/aggregator.go:180]: It’s enum

Comment (wmitsuda) [reply erigon-lib/state/aggregator.go:180]: done by: https://github.com/ledgerwatch/erigon/pull/10423

actually in this specific line the string refers to the file prefix, but I decided to extract those into a `files.go` constant file like `tables.go` for snapshot-ish constants.

but there is another places in the code in which log constants are used and there I turned into an enum method.

**Pull request merged**